script/live_content/diff_frontend_v2

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+
+# Start Content Store, Publishing API, Static, and the relevant frontend app
+# before running this script
+
+require "English"
+require "fileutils"
+require "json"
+require "net/http"
+require "uri"
+
+# For a given base path, request the frontend page with Content Store and
+# Publishing API's GraphQL endpoint as the data source and diff the results.
+
+OUTPUT_DIR = "tmp/diffs/frontend".freeze
+
+FileUtils.mkdir_p(OUTPUT_DIR)
+
+def usage
+  abort("Usage:\n\t#{$PROGRAM_NAME} a_base_path")
+end
+
+BASE_PATH = ARGV.fetch(0) { usage }
+
+def rendering_app
+  graphql_query = <<~QUERY
+    {
+      edition(base_path: "#{BASE_PATH}", content_store: "live") {
+        ...on Edition { rendering_app }
+      }
+    }
+  QUERY
+
+  response = Net::HTTP.post_form(
+    URI("http://publishing-api.dev.gov.uk/graphql"),
+    query: graphql_query,
+  )
+
+  JSON.parse(response.body).dig("data", "edition", "rendering_app")
+end
+
+def uri(with_graphql:)
+  domain = "http://#{rendering_app}.dev.gov.uk"
```

Comment:
It would be nice if we could make the domains configurable, either by env vars like:

```
frontend_host = ENV.fetch("FRONTEND_HOST", "frontend.dev.gov.uk")
```

Or with optional arguments or something. I think that would be enough to use this script against port-forwarded versions of the apps. E.g. 

```
$ k port-forward deploy/publishing-api 8080:8080 &
$ k port-forward deploy/government-frontend 8080:1337 &

$ PUBLISHING_API_HOST= localhost:8080 GOVERNMENT_FRONTEND_HOST=localhost:1337 diff_frontend_v2 ...
```

... and it would also make the lives of those of us who steadfastly refuse to use govuk-docker easier 😉

Reply:
Bruce has a PR open to do that for the JSON diffs: https://github.com/alphagov/publishing-api/pull/3740

Perhaps using an environment variable would be more flexible than hardcoding a specific port for each app